Gcc Build error: ‘function’ is not a member of ‘std’

**SLADE Version**: 5792ddf
**Operating System**: Linux

**Issue Details**:

Neither the compiler nor my IDE can find `function` as a member of `std`

```
[~] c++ --version                                                                                    
c++ (GCC) 7.1.1 20170528
```

```
[ 53%] Building CXX object src/CMakeFiles/slade.dir/Dialogs/SetupWizard/NodeBuildersWizardPage.o
/home/sergio/workspace/dev/SLADE/src/Dialogs/SScriptDialog.cpp: In member function ‘void SScriptDialog::populateScriptsTree()’:
/home/sergio/workspace/dev/SLADE/src/Dialogs/SScriptDialog.cpp:165:7: error: ‘function’ is not a member of ‘std’
  std::function<void(wxTreeCtrl*, wxTreeItemId, ArchiveTreeNode*)> addToTree =
       ^~~~~~~~
/home/sergio/workspace/dev/SLADE/src/Dialogs/SScriptDialog.cpp:165:7: note: suggested alternative: ‘is_function’
  std::function<void(wxTreeCtrl*, wxTreeItemId, ArchiveTreeNode*)> addToTree =
       ^~~~~~~~
       is_function
/home/sergio/workspace/dev/SLADE/src/Dialogs/SScriptDialog.cpp:165:64: error: expression list treated as compound expression in functional cast [-fpermissive]
  std::function<void(wxTreeCtrl*, wxTreeItemId, ArchiveTreeNode*)> addToTree =
                                                                ^
/home/sergio/workspace/dev/SLADE/src/Dialogs/SScriptDialog.cpp:165:16: error: expected primary-expression before ‘void’
  std::function<void(wxTreeCtrl*, wxTreeItemId, ArchiveTreeNode*)> addToTree =
                ^~~~
/home/sergio/workspace/dev/SLADE/src/Dialogs/SScriptDialog.cpp:187:2: error: ‘addToTree’ was not declared in this scope
  addToTree(tree_scripts_, root, scripts_dir);
  ^~~~~~~~~
[ 54%] Building CXX object src/CMakeFiles/slade.dir/Dialogs/SetupWizard/SetupWizardDialog.o
[ 54%] Building CXX object src/CMakeFiles/slade.dir/Dialogs/SetupWizard/TempFolderWizardPage.o
make[2]: *** [src/CMakeFiles/slade.dir/build.make:1703: src/CMakeFiles/slade.dir/Dialogs/SScriptDialog.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:93: src/CMakeFiles/slade.dir/all] Error 2
make: *** [Makefile:128: all] Error 20
```

--- a/src/MapEditor/SLADEMap/SLADEMap.cpp
+++ b/src/MapEditor/SLADEMap/SLADEMap.cpp
@@ -3053,7 +3053,7 @@ vector<fpoint2_t> SLADEMap::cutLines(double x1, double y1, double x2, double y2)
 		{
 			// Add intersection point to vector
 			intersect_points.push_back(intersection);
-			LOG_DEBUG("Intersection point", intersection, "valid with", lines[a]);
+			LOG_DEBUG("Intersection point", intersection, "valid with", lines_[a]);
 		}
 		else if (intersection != cutter.p1())
 		{

--- a/src/common.h
+++ b/src/common.h
@@ -141,6 +141,7 @@
 // C++
 #include <map>
 #include <vector>
+#include <functional>
 #include <algorithm>
 #include <set>
 #include <cmath>

